### ISSUE TITLE
[Feature] Allow to set the default Locations & Symbol-type per file [sf#239]

### ISSUE THREAD

**Issue by openMSX-import** [openMSX/debugger#38]
**Reported by sd-snatcher on 2012-09-18 23:00 UTC**

Currently, the "Locations" (slot/subslot) and "Symbol type" (Jump label, Variable label or value) can only be set on a per-label basis on the "Address Label" tab. For hundreds of labels like the MSX system has this is not very productive and end up not being used.

Please add an option to set the default "Locations" and "Symbol-type" per file on the "Symbols file" tab of the "Symbol Manager" window. This will set the default options for all labels in that file, then the programmer can fine-tune on the "Address labels" tab only the ones he needs. This setup matches better a real world scenario, when the programmer would have a set of SYM files like this:

(locations used a Panasonic WSX as an example)

MSXBIOS.sym   (location: slot=0.0, Symbol-type=Jump)
MSXhooks.sym  (location: slot=3.0, Symbol-type=Jump)
MSXvarsROM.sym (location: slot=0.0, Symbol-type=Variable)
MSXvarsRAM.sym (location: slot=3.0, Symbol-type=Variable)
MyROMgame.sym  (location: slot=1, Symbol-type=Jump/Variable)

**Comment by m9710797** [comment]
BTW the "Symbol type" could be set, but the (old) debugger didn't use that information in any way, so it was actually useless to set it. The new debugger doesn't have this field anymore.

In the new debugger the slot can only be set per symbol-file rather than on individual symbols. That's already more in line with this feature request (although now it's the only possibility rather than the default).

In general openMSX currently doesn't allow to edit symbols. E.g. you can't manually add symbols, or change the value or name. Maybe we could add a symbol editor (or extend the current symbol viewer). On the other hand, editing the symbol file directly (in an external editor) is very easy, and then you also have the symbols in a more widely re-usable format. Reloading symbol files in openMSX is already very easy.

So to summarize: the new debugger does not have this, but IMHO is also not needed (the workaround is easy and better).

**Comment by MBilderbeek** [comment]
Alright, let's close this for now. If people disagree, feel free to open a new issue in the openMSX/openMSX tracker.
